packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -978,6 +990,121 @@ describe('TransactionController', () => {
       },
     );
 
+    it('updates transaction if type is swap approval', async () => {
```

Comment:
We could test all this logic in isolation if we move to `utils/swaps.ts` which may be simpler.

Reply:
It's all in `swaps.ts` now https://github.com/MetaMask/core/pull/1877/commits/127444c3b779230c85656050ea48613c3ff3de71